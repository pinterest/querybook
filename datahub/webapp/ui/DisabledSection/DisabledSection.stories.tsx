import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import centered from '@storybook/addon-centered/react';

import { DisabledSection } from './DisabledSection';
import { DebouncedInput } from 'ui/DebouncedInput/DebouncedInput';

export default {
    title: 'Stateless/DisabledSection',
    decorators: [centered],
};
export const _DisabledSection = () => (
    <DisabledSection>
        <DebouncedInput
            flex
            inputProps={{
                placeholder: 'placeholder',
                className: 'input',
                type: 'text',
            }}
            onChange={() => null}
            value="disabled input"
        />
    </DisabledSection>
);
