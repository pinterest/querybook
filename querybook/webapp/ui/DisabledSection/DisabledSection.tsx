import styled from 'styled-components';

interface IDisabledSectionProps {
    disabled: boolean;
}

export const DisabledSection = styled.div<IDisabledSectionProps>`
    ${({ disabled = true }: IDisabledSectionProps) =>
        disabled &&
        `
    cursor: no-drop;
    pointer-events: none;
`};
`;
