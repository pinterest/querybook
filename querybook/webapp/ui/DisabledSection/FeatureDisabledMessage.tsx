import React from 'react';

import { Icon } from 'ui/Icon/Icon';
import { Message } from 'ui/Message/Message';

interface FeatureDisabledMessageProps {
    message?: string;
}

export const FeatureDisabledMessage: React.FunctionComponent<
    FeatureDisabledMessageProps
> = ({ message = 'This feature is currently disabled.' }) => (
    <div className="feature-disabled">
        <Icon
            name="AlertCircle"
            size={128}
            color="light"
            className="feature-disabled-icon"
        />
        <Message message={message} type="info" />
    </div>
);
