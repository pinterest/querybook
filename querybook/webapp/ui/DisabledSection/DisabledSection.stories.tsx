import centered from '@storybook/addon-centered/react';
import React from 'react';

import { DebouncedInput } from 'ui/DebouncedInput/DebouncedInput';

import { DisabledSection } from './DisabledSection';

export default {
    title: 'Stateless/DisabledSection',
    decorators: [centered],
};
export const _DisabledSection = () => (
    <DisabledSection>
        <DebouncedInput
            flex
            inputProps={{
                placeholder: 'placeholder',
                className: 'input',
                type: 'text',
            }}
            onChange={() => null}
            value="disabled input"
        />
    </DisabledSection>
);
