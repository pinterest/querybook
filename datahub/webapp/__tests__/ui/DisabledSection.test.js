import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';

import { DisabledSection } from '../../ui/DisabledSection/DisabledSection';

it('renders without crashing', () => {
    shallow(
        <DisabledSection>
            <input />
        </DisabledSection>
    );
});

describe('matches enzyme snapshots', () => {
    it('matches snapshot', () => {
        let wrapper = shallow(
            <DisabledSection>
                <input />
            </DisabledSection>
        );
        let serialized = toJson(wrapper);
        expect(serialized).toMatchSnapshot();
    });
    it('matches snapshot - false', () => {
        let wrapper = shallow(
            <DisabledSection disabled={false}>
                <input />
            </DisabledSection>
        );
        let serialized = toJson(wrapper);
        expect(serialized).toMatchSnapshot();
    });
});

describe('matches test renderer snapshot', () => {
    it('serializes the styles', () => {
        const output = renderer.create(
            <DisabledSection>
                <input />
            </DisabledSection>
        );
        expect(output).toMatchSnapshot();
    });
});
