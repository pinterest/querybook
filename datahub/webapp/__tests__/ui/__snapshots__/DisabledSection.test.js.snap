// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches enzyme snapshots matches snapshot - false 1`] = `
<StyledComponent
  disabled={false}
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
    ",
          [Function],
          ";
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  <input />
</StyledComponent>
`;

exports[`matches enzyme snapshots matches snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
    ",
          [Function],
          ";
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  <input />
</StyledComponent>
`;

exports[`matches test renderer snapshot serializes the styles 1`] = `
<div
  className="sc-bdVaJa qzEVL"
>
  <input />
</div>
`;
